File: API/Models/Guardian.cs
using System;
using System.Text.Json.Serialization;

namespace API.Models;

public class Guardian
{
   public int Id { get; set; }
   public required string FirstName { get; set; }
   public string? SecondName { get; set; }
   public required string LastName { get; set; }
   public required string Gender { get; set; }
   public required string Relationship { get; set; }
   public required string IdentityNumber { get; set; }
   public string? Email { get; set; }
   public required string Phone { get; set; }
   public required DateOnly DateOfBirth { get; set; }
   //Address
   public int AddressId { get; set; }
   public required Address Address { get; set; }
   //Students (one-to-many)
   [JsonIgnore]//stop JSON serialization from recursing endlessly.
   public ICollection<Student> Students { get; set; } = [];
}

File: Client/src/app/_interfaces/guardian.ts
import { Address } from "./address";
import { Student } from "./student";

export interface Guardian {
  id: number;
  firstName: string;
  secondName?: string;
  lastName: string;
  gender: string;
  relationship: string;
  identityNumber: string;
  email?: string;
  phone: string;
  dateOfBirth: string;
  addressId: number;
  Address: Address
  Students: Student[];
}
